src/commands/build-util.ts

```diff
@@ -44,70 +46,52 @@ export async function buildFile(
       continue;
     }
     const staleOutputs = new Set<string>();
-    for (const destPath of Object.keys(output)) {
+    for (const destExt of Object.keys(output)) {
       // If the current output file extension doesn't match a plugin input, skip it.
-      if (!step.input.includes(getExt(destPath).baseExt)) {
+      if (!step.input.includes(destExt)) {
         continue;
       }
-      const destBuildFile = output[destPath];
+      const destBuildFile = output[destExt];
       const result = await step.build({
-        contents: destBuildFile.contents,
-        filePath: destPath,
+        contents: destBuildFile,
+        filePath: rootFileName + destExt,
```

Comment:
important question: are we okay with filePath just being file name? As written it rarely points to an actual file on disk, and today we mainly use it for file extension checking.

If worried about backwards compatibility, we could keep the current fake `filePath` and add a new `fileName` option, removing `filePath` in a future release.

Reply:
I’d be +1 in favor of `filename` being the … filename. And deprecating `filePath`.

I think the only reason I kept it was for the Babel CLI, really (build scripts). You’re right that it doesn’t really make sense going forward.